caffe2/serialize/inline_container.h

```diff
@@ -59,195 +63,271 @@ namespace torch { namespace jit {
+      return false;
+    }
+    // magic number mismatch in PyTorch file.
+    uint64_t file_format_version = read64BitIntegerLittleEndian();
+    if (file_format_version > kMaxSupportedFileFormatVersion) {
+      LOG(ERROR) << "Attempted to read a PyTorch file with version "
+        << file_format_version
+        << " but the maximum supported version for reading is "
+        << kMaxSupportedFileFormatVersion
+        << ". Your PyTorch installation may be too old.";
+      return false;
+    }
+    seekToNextAlignmentBoundary();
+    return true;
+  }
+
+  bool readAndValidateFileFooter() {
     // Seek to location of file footer. We've already validated that the file
     // length is a multiple of the alignment size
-    cursor = file_size - kFieldAlignment;
-    in.seekg(cursor);
+    cursor_ = file_size_ - kFieldAlignment;
+    in_->seekg(cursor_);
     auto tag = read64BitIntegerLittleEndian();
     if (tag != RecordTags::FOOTER) {
-      throw std::runtime_error("File footer has wrong record type. Is this"
-                               " file corrupted?");
+      LOG(ERROR) << "File footer has wrong record type. Is this"
+        << " file corrupted?";
+      return false;
     }
-    last_record_offset = read64BitIntegerLittleEndian();
-    if (last_record_offset > file_size) {
-      throw std::runtime_error("Offset of last record is higher than the size"
-                               " of the file! Is this file corrupted?");
+    last_record_offset_ = read64BitIntegerLittleEndian();
+    if (last_record_offset_ > file_size_) {
+      LOG(ERROR) << "Offset of last record is higher than the size"
+        << " of the file! Is this file corrupted?";
+      return false;
     }
+    return true;
   }
 };
 
-class PyTorchStreamWriter {
+class PyTorchStreamWriter final {
  public:
-  PyTorchStreamWriter(std::ostream& out_) : out(out_) {
+  PyTorchStreamWriter(std::ostream* out) : out_(out) {
+    pad_buffer_.resize(kFieldAlignment, kPadValue);
     writeFileHeader();
     // In the case that we do not write any records into this file, the last
     // record index written into the footer will point to the footer itself.
-    last_record_idx = cursor;
+    last_record_idx_ = cursor_;
   }
-  uint64_t writeRecord(const char* data, size_t size) {
-    JIT_ASSERT(!finalized);
-    uint64_t record_offset = cursor;
-    last_record_idx = record_offset;
+
+  uint64_t writeRecord(const void* data, size_t size) {
+    CAFFE_ENFORCE(!finalized_, "should not be finalized!");
+    uint64_t record_offset = cursor_;
+    last_record_idx_ = record_offset;
     write64BitIntegerLittleEndian(RecordTags::STORAGE);
     write64BitIntegerLittleEndian(size);
     padToNextAlignmentBoundary();
     writeBuffer(data, size);
     padToNextAlignmentBoundary();
     return record_offset;
   }
+
   void writeEndOfFile() {
-    JIT_ASSERT(!finalized);
+    CAFFE_ENFORCE(!finalized_, "cannot finalize again!");
```

Comment:
so are you using CAFFE_ENFORCE/CAFFE_THROW or runtime_error or AT_ASSERT/AT_ERROR - let's be consistent. For now either CAFFE_ENFORCE or AT_ASSERT is fine (they are merging)